### ISSUE TITLE
Bootsnap Cache Growing Very Large

### ISSUE THREAD

**Issue by cookrn** [Shopify/bootsnap#153]
I've begun noticing a slowdown during a certain step of a deploy process that ensures file permissions and traced it to the Bootsnap cache in the tmp directory.

Locally my cache size is 301M, but in some deployed environments, it's as large as 7.8G and is 6.4G on the one being looked at this moment. The command I'm using to judge the size of the cache is `du -sh tmp/cache/bootsnap-compile-cache`

The number of files in the cache is very different as well where I'm seeing ~18K locally in the cache, but ~430K in the deployed environment. This is found using `ls -alh tmp/cache/bootsnap-compile-cache/* | wc -l`.

Gem Versions:
* Rails Version: 4.2.10
* Bootsnap Version: 1.1.7

Is it possible that the changing of permissions is causing the issue? It's not easy/possible for that to be disabled in my case, so I could look at just deleting the cache upon deploy. Are there any other potential avenues to look down or hints of things-done-wrong based on others' production experience?

Thanks in advance!

(Note: I will look into upgrading the `bootsnap` gem as we don't have a hard reason to be locked at the version mentioned above.)

**Comment by x-yuri** [comment]
@burke Should I delete both `tmp/cache/bootsnap-compile-cache` and `tmp/cache/bootsnap-load-path-cache`? In [my case][1] it was:

[1]: https://github.com/Shopify/bootsnap/issues/185#issuecomment-412552950

```
59G     tmp/cache/bootsnap-compile-cache
231M    tmp/cache/bootsnap-load-path-cache
```

How do you decide when it's time to clean the cache?

**Comment by burke** [comment]
Yeah I'd say you should delete those. I don't have a great heuristic for when it's a good idea. Our container build process throws them away weekly just as an accident of the way it's built; that seems to work fine.
